sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2110,4 +2112,28 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       checkAnswer(res, Row("1-1", 6, 6))
     }
   }
+
+  test("SPARK-26572: fix aggregate codegen result evaluation") {
```

Comment:
Since this is a problem with whole stage codegen, waht about moving this test to `WholeStageCodegenSuite`? And adding an assert that whole stage codegen is actually used, ie. the HashAggregate is a child of WholeStageCodegenExec?

Reply:
I'm fine with moving it to `WholeStageCodegenSuite` but the plan looks like:
```
*(3) Project [idx#4, id#6L]
+- *(3) BroadcastHashJoin [idx#4], [idx#9], Inner, BuildRight
   :- *(3) HashAggregate(keys=[idx#4], functions=[], output=[idx#4, id#6L])
   :  +- Exchange hashpartitioning(idx#4, 1)
   :     +- *(1) HashAggregate(keys=[idx#4], functions=[], output=[idx#4])
   :        +- *(1) Project [value#1 AS idx#4]
   :           +- LocalTableScan [value#1]
   +- BroadcastExchange HashedRelationBroadcastMode(List(cast(input[0, int, false] as bigint)))
      +- *(2) Project [value#1 AS idx#9]
         +- LocalTableScan [value#1]
```
so I guess you mean checking `WholeStageCodegenExec` has a `ProjectExec` child that has a `BroadcastHashJoinExec` child?

Reply:
Moved and added `WholeStageCodegenExec` check.